src/Features/LanguageServer/Protocol/Handler/Diagnostics/Public/PublicDocumentPullDiagnosticsHandler.cs

```diff
@@ -2,52 +2,42 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System;
 using System.Collections.Immutable;
 using System.Linq;
-using System.Threading;
-using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.Diagnostics;
-using Microsoft.CodeAnalysis.EditAndContinue;
+using Microsoft.CodeAnalysis.LanguageServer.Handler.Diagnostics.DiagnosticSources;
 using Microsoft.CodeAnalysis.Options;
 using Roslyn.LanguageServer.Protocol;
-using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.LanguageServer.Handler.Diagnostics.Public;
 
-using DocumentDiagnosticReport = SumType<RelatedFullDocumentDiagnosticReport, RelatedUnchangedDocumentDiagnosticReport>;
-
 // A document diagnostic partial report is defined as having the first literal send = DocumentDiagnosticReport (aka changed / unchanged) followed
 // by n DocumentDiagnosticPartialResult literals.
-// See https://github.com/microsoft/vscode-languageserver-node/blob/main/protocol/src/common/proposed.diagnostics.md#textDocument_diagnostic
+// See https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#documentDiagnosticParams
 using DocumentDiagnosticPartialReport = SumType<RelatedFullDocumentDiagnosticReport, RelatedUnchangedDocumentDiagnosticReport, DocumentDiagnosticReportPartialResult>;
+using DocumentDiagnosticReport = SumType<RelatedFullDocumentDiagnosticReport, RelatedUnchangedDocumentDiagnosticReport>;
 
 [Method(Methods.TextDocumentDiagnosticName)]
 internal sealed partial class PublicDocumentPullDiagnosticsHandler : AbstractDocumentPullDiagnosticHandler<DocumentDiagnosticParams, DocumentDiagnosticPartialReport, DocumentDiagnosticReport?>
 {
-    private readonly string _nonLocalDiagnosticsSourceRegistrationId;
     private readonly IClientLanguageServerManager _clientLanguageServerManager;
 
     public PublicDocumentPullDiagnosticsHandler(
         IClientLanguageServerManager clientLanguageServerManager,
         IDiagnosticAnalyzerService analyzerService,
+        IDiagnosticSourceManager diagnosticSourceManager,
         IDiagnosticsRefresher diagnosticsRefresher,
         IGlobalOptionService globalOptions)
-        : base(analyzerService, diagnosticsRefresher, globalOptions)
+        : base(analyzerService, diagnosticsRefresher, diagnosticSourceManager, globalOptions)
     {
-        _nonLocalDiagnosticsSourceRegistrationId = Guid.NewGuid().ToString();
         _clientLanguageServerManager = clientLanguageServerManager;
     }
 
-    /// <summary>
-    /// Public API doesn't support categories (yet).
-    /// </summary>
-    protected override string? GetDiagnosticCategory(DocumentDiagnosticParams diagnosticsParams)
-        => null;
+    protected override string? GetRequestDiagnosticCategory(DocumentDiagnosticParams diagnosticsParams)
+        => diagnosticsParams.Identifier;
```

Comment:
i'm trying to figure out if htis cahnge was intentional.  is this because the public spec now supports categories?  and this is how it is passed in?  do we test this?

Reply:
@dibarbet  as well for attention here.

Reply:
yes, identifier is the public spec version of categories.  intentional

Reply:
@dibarbet made this change; I ported it from his branch